### PR TITLE
Enhancing Interactor

### PR DESCRIPTION
**Pull request by hkchekc**
This Pull Request add some additional functions that is potentially useful for using the Interactive Plotting:

- Gives common indexing for all objects (mpl.Artist) for easier future Object-Orientated Development
- Add Sliders for zooming on the x-axis
- Hide non-referenced DataArrays when areas are tagged
- Fix bug regarding legends and allow users to show/hide legends
- Add overview functions for users to quickly know what can (likely) be plotted together

### CODE REVIEW

File: nixworks/plotter/plotter.py [hunk 1]
```diff
@@ -160,12 +160,17 @@ def plot(self, axis=None):
     def plot_1d(self):
         data = self.array[:]
         if self.array.dimensions[self.xdim].dimension_type == nix.DimensionType.Set:
+            print('abc')
             categories = list(self.array.dimensions[self.xdim].labels)
+            print(categories)
+            print()
```
Comment (achilleas-k) [inline nixworks/plotter/plotter.py]: Leftover debug prints?

**Comment by jgrewe** [comment]
apparently, @achilleas-k does not have the same issue, but I cannot run the notebook. 
```NotJSONError('Notebook does not appear to be JSON: \'{\\n "cells":```

**Comment by achilleas-k** [comment]
> apparently, @achilleas-k does not have the same issue, but I cannot run the notebook.
> `NotJSONError('Notebook does not appear to be JSON: \'{\\n "cells":`

I have the same issue.

File: nixworks/plotter/interactor.py [hunk 2]
```diff
@@ -3,66 +3,94 @@
...
         # List for labels
         self.labels= {'x': None, 'y': None}
 
-    def _plot_da(self, data_arrays, enable_tag_list =False):
+    @staticmethod
+    def _check_da_combination(data_arrays):
+        # checking if the DataArrays can be put in the same graph
+        u = data_arrays[0].unit
+        if not util.units.is_si(u):
+            raise ValueError("Invalid Unit")
```
[the remark this answers is not in the document]
Comment (hkchekc) [reply nixworks/plotter/interactor.py]: Will fix in the next one

**Comment by achilleas-k** [comment]
@jgrewe Should I merge?

**Comment by jgrewe** [comment]
oh, yes, pleas go ahead

**Pull request merged**